### DISCUSSION TITLE
Request for Metadata in Rafiki Webhook Responses

### DISCUSSION THREAD

**Discussion by CollinsMunene** [interledger/rafiki#2952] [Integrator Feedback]
Hi, this is Collins from Vuza Finance and I have been closely monitoring the webhooks that an Account Service Extension (ASE) receives from Rafiki for both incoming and outgoing events. In these webhooks, the response body does not appear to include an option for metadata. In some of the applications I am currently integrating with Rafiki, it would be highly beneficial to retrieve metadata passed during transaction creation.

For example, when creating a successful outgoing payment for a loan transaction and subsequently executing a successful DepositOutgoingPaymentLiquidity, it would be helpful to pass metadata such as the loan ID within the mutation. This would allow the metadata to be returned in the subsequent webhooks. This way, if I need to update a record in my system, I can retrieve the loan ID from the metadata and process the transaction further.

Could you please confirm whether this functionality exists or if it is planned? I have not encountered it in the current implementation.

**Comment by mkurapov** [comment]
Hi @CollinsMunene!

Currently, the webhooks received by the ASE do include `metadata` in the outgoing/incoming payment fields, _if_ the `metadata` was provided in the original creation request for said payment.

Example webhook page in the Admin UI:

![Screenshot 2024-09-09 at 19 40 21](https://github.com/user-attachments/assets/0f35495e-2de9-472f-adad-08c6ab6c9f38)

What I'm understanding is, is that you would like to be able to add additional metadata to the existing resources at any point of the payment, is that correct? (For example, when withdrawing or depositing payment liquidity).

As of last week, you are able to use `updateIncomingPayment` [mutation](https://github.com/interledger/rafiki/issues/2892) to do just that: add/override metadata for an incoming payment at your Rafiki instance. Would you like to be able to do the same for outgoing payments?

**Reply by CollinsMunene** [reply]
Hi @mkurapov, this is well noted. I have retested the process again. The metadata used to create the incoming and outgoing payments is sufficient.

I would like to be able to update the metadata of an outgoing payment in the case scenario where we want to give the internal team the capability of updating such records during book balancing( can update the description metadata to clearly label the transaction and such).
